src/aleph/handlers/message_handler.py

```diff
@@ -407,6 +405,22 @@ async def process(
             )
             return ProcessedMessage(message=existing_message, is_confirmation=True)
 
+        # Note: Check if message is already forgotten (and confirm it)
+        # this is to avoid race conditions when a confirmation arrives after the FORGET message has been preocessed
+        forgotten_message = get_forgotten_message(
+            session=session, item_hash=ItemHash(pending_message.item_hash)
+        )
+        if forgotten_message:
+            await self.confirm_existing_forgotten_message(
+                session=session,
+                forgotten_message=forgotten_message,
+                pending_message=pending_message,
+            )
+            return RejectedMessage(
+                pending_message=pending_message,
+                error_code=ErrorCode.FORGET_TARGET_NOT_FOUND,
```

Comment:
I think that we need to create a specific error code for that situation.

Reply:
I have created a new code called `FORGET_FORGOTTEN` for this scenario